### PR TITLE
Add `overline` style and remove `keyword`, `hsl`, `hsv`, `hwb` and `ansi` color spaces

### LINKED ISSUE #360: Remove some color types
HCG was added after `color-convert` was added to ansi-styles. We should officially support it in `ansi-styles` and document its support here. It's similar to HSL and is (slowly) increasing in popularity.

### PR DESCRIPTION
**Pull request by Richienb**
# Update `ansi-styles` package
- Add `overline` style
- Remove `keyword`, `hsl`, `hsv`, `hwb` and `ansi` color spaces

#431 

Fixes #360

### CODE REVIEW

**Comment by Richienb** [comment]
Because desampling to `ansi` is no longer active, level 1 is now effectively the same in function as level 2.

**Comment by Qix-** [comment]
Just curious, why aren't we downsampling anymore? That seems like a functional regression, given that there are still _loads_ of terminals that only support the older 16 color palette.

**Comment by Richienb** [comment]
Because you said so

**Comment by Qix-** [comment]
I don't see in there why we need to remove this functionality. I would say we should inline that from `color-convert` (you're free to copy over the `ansi256.ansi16` function from `conversions.js` directly - not that the license would have prevented you anyway). Just mention it came from `color-convert` in a comment.

The concern about upscaling (#370) was valid, but downscaling is absolutely necessary to keep compatibility with people who do not have 256 support.

**Comment by Richienb** [comment]
> you're free to copy over the `ansi256.ansi16` function from `conversions.js` directly - not that the license would have prevented you anyway

That function doesn't exist

**Comment by Qix-** [comment]
Right, you have to merge them - this is untested but should be close enough. Just needs a bit of polish for the Chalk conventions.

```javascript
function ansi256To16(args) {
	let r, g, b;

	// Handle greyscale
	if (args >= 232) {
		const c = (args - 232) * 10 + 8;
		r = g = b = c / 255;
	} else {
		args -= 16;

		let rem;
		r = Math.floor(args / 36) / 5;
		g = Math.floor((rem = args % 36) / 6) / 5;
		b = (rem % 6) / 5;
	}

        const value = Math.max(r, g, b) * 2;

	if (value === 0) {
		return 30;
	}

	let ansi = 30
		+ ((Math.round(b) << 2)
		| (Math.round(g) << 1)
		| Math.round(r));

	if (value === 2) {
		ansi += 60;
	}

	return ansi;
}
```

**Comment by Richienb** [comment]
Support for `ansi` has been added back in.

**Review by Qix- — APPROVED** [review]
I don't see anything glaringly wrong with the code; I'll let Sindre have the final look.

**Comment by sindresorhus** [comment]
I'm confused. I understand preserving the downsampling, but shouldn't we still remove the explicit `.ansi()` method?

**Comment by Richienb** [comment]
IMO, there is no impact of leaving the function there so we might as well leave it for people wanting to use the `ansi` color space.

**Comment by Qix-** [comment]
People wanting to use the `ansi` color space should just use `.red()`, `.blueBright()`, etc as it's a 1:1 mapping. People doing programmatic generation of colors for whatever reason are going to be using 256 or truecolor - not 16 - as there is no logical rhyme or reason to the code (I always remember them as "traffic light, dark blue, pink, light blue, white" - completely arbitrary).

I agree with sindre here - no need for `.ansi()`. Chalk is a utility library, after all - you're always able to write escapes yourself with trivial syntax.

**Comment by Richienb** [comment]
Removed

**Review by sindresorhus — APPROVED** [review]

File: index.d.ts [hunk 1]
```diff
@@ -178,51 +184,14 @@ export interface ChalkInstance extends ChalkFunction {
...
-	Use HWB values to set text color.
-	*/
-	hwb: (hue: number, whiteness: number, blackness: number) => this;
-
-	/**
-	Use a [Select/Set Graphic Rendition](https://en.wikipedia.org/wiki/ANSI_escape_code#SGR_parameters) (SGR) [color code number](https://en.wikipedia.org/wiki/ANSI_escape_code#3/4_bit) to set text color.
-
-	30 <= code && code < 38 || 90 <= code && code < 98
-	For example, 31 for red, 91 for redBright.
+	Use a [8-bit unsigned number](https://en.wikipedia.org/wiki/ANSI_escape_code#8-bit) to set text color.
```
Comment (Qix-) [inline index.d.ts:187]:
```suggestion
	Use an [8-bit unsigned number](https://en.wikipedia.org/wiki/ANSI_escape_code#8-bit) to set text color.
```

**Review by Qix- — CHANGES_REQUESTED** [review]

**Review by Qix- — APPROVED** [review]
LGTM

**Comment by sindresorhus** [comment]
Nice work, @Richienb 👍

**Comment by Qix-** [comment]
Yes indeed :) Thanks @Richienb, super appreciated.

**Pull request merged**